### PR TITLE
[build] bootstrap libtool dependency documentation fix

### LINKED ISSUE #2026: ./bootstrap fails with error "not found"
Hi folks,

Following the instructions for building the CC2650 driver demo, I've installed autoconf, automake, and the gcc-arm toolchain. After running `./bootstrap` from the repo root, I get this error:

```
./third_party/nlbuild-autotools/repo/scripts/bootstrap: 280: ./third_party/nlbuild-autotools/repo/scripts/bootstrap: --automake: not found
```

Have I just missed a set-up step somewhere?

### PR DESCRIPTION
**Pull request by nealjack**
Add instructions for installing libtool to all example READMEs. Add check in bootstrap for the presence of libtool. Fixes #2026.

### CODE REVIEW

**Comment by nealjack** [comment]
I signed the CLA.

**Comment by jwhui** [comment]
Thanks for this submission!  We certainly need to make the libtool dependency obvious to the developer.  However, this PR assumes the presence of `apt-get`.

There are a couple of things we could do to address the problem:
1. Provide a GNU toolchain setup guide that lists required packages.
2. Adopt the [Scripts To Rule Them All](https://github.com/github/scripts-to-rule-them-all) approach that performs the necessary setup for each development platform.

We should probably do both.  Thoughts?

**Comment by nealjack** [comment]
I'm in agreement. I can incorporate the second suggestion into my changes to the bootstrap script. Is there a reason the scripts are not in a script directory?

File: Aptfile [hunk 1]
```diff
@@ -0,0 +1 @@
+libtool
```
Comment (jwhui) [inline Aptfile]: Let's just inline this for now, rather than creating a new top-level file.

File: bootstrap [hunk 2]
```diff
@@ -42,4 +42,9 @@ srcdir=`dirname ${0}`
 abs_srcdir=`pwd`
 abs_top_srcdir="${abs_srcdir}"
 
+# install Debian flavor dependencies
+if [ -f "Aptfile" ] && $(apropos "package manager" | grep -wq dpkg) ; then
```
Comment (jwhui) [inline bootstrap]: Should we just test for `apt-get` here, similar to what we did in https://github.com/openthread/borderrouter/blob/master/script/bootstrap?

File: examples/apps/cli/README.md [hunk 3]
```diff
@@ -10,6 +10,7 @@ another emulated Thread device.
 
 ```bash
 $ cd <path-to-openthread>
+$ sudo apt-get install libtool
```
Comment (jwhui) [inline examples/apps/cli/README.md]: These doc modifications should no longer be needed with the above changes.

**Comment by nealjack** [comment]
Out of the dependencies documented in the [codelab](https://codelabs.developers.google.com/codelabs/openthread-simulation/index.html#1), which are required for this repo specifically besides libtool?

**Comment by jwhui** [comment]
> Out of the dependencies documented in the [codelab](https://codelabs.developers.google.com/codelabs/openthread-simulation/index.html#1), which are required for this repo specifically besides libtool?

Depends on what the user is trying to do.  To just compile the posix example and run the simulation, only automake and libtool packages are needed.  The autoconf package is a dependency of automake, so no need to install that separately.  Of course, gcc or clang is also needed, but most linux distributions come pre-installed with that.

Thinking about this further, I'm not sure we should combine the development toolchain setup together with OpenThread's build system setup.  I propose we put your changes in `script/bootstrap` and update the `Toolchain` section of the example READMEs.  Thoughts?

**Comment by jwhui** [comment]
@nealjack, thoughts on my last comment?

**Comment by nealjack** [comment]
@jwhui I made changes in line with your thoughts. `bootstrap` now only installs automake and libtool, and I've included directions for installing the GNU toolchain in each of the example READMEs for platforms with Apt.

**Comment by jwhui** [comment]
Thanks for the update.  Unfortunately, I think my latest proposal wasn't clear.  Let me try to reiterate:
1. Keep the existing `bootstrap` script unmodified, this is used for creating the appropriate Makefiles.
2. Create a new `script/bootstrap` script that includes installing package dependencies.
3. Update the `Toolchain` section of each `README.md` to include the `./script/bootstrap` command for installing package dependencies.

Thoughts?

**Comment by nealjack** [comment]
Ok, so do you want all of my changes to `bootstrap` to be reverted? This isn't ideal because the original problem was that boostrap did not install the automake and libtool dependencies and failed on a vague error message. I think the highest level bootstrap should install libtool and automake before everything, and the addition of a `script/bootstrap` should install all toolchain dependencies needed for compilation of the examples.
Are we thinking of the same changes here?

**Comment by jwhui** [comment]
I agree that the vague error message is not ideal.  I believe this should be addressed in `nlbuild-autotools` directly.

The top-level `./bootstrap` follows existing convention with autotools.
- [Autotools FAQ](https://www.gnu.org/software/automake/faq/autotools-faq.html#What-does-_002e_002fbootstrap-or-_002e_002fautogen_002esh-do_003f)
- [GNU Autoconf, Automake and Libtool](https://sourceware.org/autobook/autobook/autobook_43.html)

The use of `script/bootstrap` follows the [Scripts To Rule Them All](https://github.com/github/scripts-to-rule-them-all), which serves a different purpose.

I also agree the file naming is not ideal and is an unfortunate side-effect of trying to adopt two different conventions (i.e. GNU Autotools and Script to Rule Them All).  Maybe we use `script/toolchain-setup` instead.

**Comment by nealjack** [comment]
Regardless, at some point there needs to be documentation indicating that libtool/autotool needs to be installed (either through first running `script/bootstrap`  or installing them manually). Should this be indicated in the example READMEs?

File: examples/platforms/cc2538/README.md [hunk 4]
```diff
@@ -22,6 +22,7 @@ In a Bash terminal, follow these instructions to build the cc2538 examples.
 
 ```bash
 $ cd <path-to-openthread>
+$ ./script/bootstrap
```
Comment (jwhui) [inline examples/platforms/cc2538/README.md]: Should we move this line into the `Toolchain` section above?

**Comment by jwhui** [comment]
Yes, I believe it makes sense to note `script/bootstrap` in the READMEs.  That's what was intended by my Item 3 in a previous comment :)

File: examples/platforms/cc2538/README.md (hunk 4 above)
Comment (nealjack) [inline examples/platforms/cc2538/README.md]: Yeah, that makes sense. I noticed that there is some inconsistencies in the format and content of these example READMEs (i.e. the cc256x vs nrf52840), even though they depend on the same toolchain. Should we work to make them cohesive and identical?

File: examples/platforms/da15000/README.md [hunk 5]
```diff
@@ -4,12 +4,42 @@ This directory contains example platform drivers for the [Dialog Semiconductor D
...
+default on modern Posix systems. Windows does not have these tools installed by
+default, and the bootstrap script requires a Posix or MSYS environment to run.
+It is possible to setup an MSYS environment inside of Windows using tools such
+as [Cygwin][cygwin] or [MinGW][mingw] but it is recommended to setup a Linux VM
+for building on a Windows system. For help setting up VirtualBox with Ubuntu,
+consult this [community help wiki article][ubuntu-wiki-virtualbox].
+
+[gnu-autoconf]: https://www.gnu.org/software/autoconf
+[gnu-automake]: https://www.gnu.org/software/automake
+[python]: https://www.python.org
```
Comment (jwhui) [inline examples/platforms/da15000/README.md]: The python requirement was specifically for `cc2538-bsl.py` using the CC25xx platforms.

File: examples/platforms/emsk/README.md [hunk 6]
```diff
@@ -19,9 +19,33 @@ The EMSK platform is a configurable, FPGA-based software development platform fo
 
 ## Toolchain
 
-Download and install [GNU toolchain for ARC EM][gnu-toolchain].
+Building the examples for the emsk requires [GNU AutoConf][gnu-autoconf],
+[GNU AutoMake][gnu-automake], [Python][python], and the
+[ARM gcc toolchain][arm-toolchain].
```
Comment (jwhui) [inline examples/platforms/emsk/README.md]: This platform is not ARM-based.

File: examples/platforms/gp712/README.md [hunk 7]
```diff
@@ -4,15 +4,38 @@ This directory contains example platform drivers for Qorvo gp712 on RPi.
 
 ## Toolchain
 
-This example use the GNU GCC toolchain on the Raspberry Pi.
-To build on the Pi:
-1) Download the repo to the Pi
-2) go to the subfolder in the openthread repo: third_party/nlbuild-autotools/repo/tools/packages and enter this command:
+Building the examples for the gp712 requires [GNU AutoConf][gnu-autoconf],
+[GNU AutoMake][gnu-automake], [Python][python], and the
+[ARM gcc toolchain][arm-toolchain].
```
Comment (jwhui) [inline examples/platforms/gp712/README.md]: This platform runs on raspberry pi.

**Comment by nealjack** [comment]
Ok, I just added minimal instructions to the relevant platforms.

File: script/bootstrap [hunk 8]
```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
+#
+#  Copyright (c) 2016, The OpenThread Authors.
```
Comment (jwhui) [inline script/bootstrap]: nit: 2017

File: script/bootstrap [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+#    Description:
+#      This file is a trampoline script to the nlbuild-autotools
```
Comment (jwhui) [inline script/bootstrap]: This description does not apply to this file.

**Pull request merged**